src/SlugUpdateHandler/SlugUpdateHandler.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusRedirectPlugin\SlugUpdateHandler;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Setono\SyliusRedirectPlugin\Exception\SlugUpdateHandlerValidationException;
+use Setono\SyliusRedirectPlugin\Factory\RedirectFactoryInterface;
+use Setono\SyliusRedirectPlugin\Finder\RemovableRedirectFinderInterface;
+use Sylius\Component\Channel\Model\ChannelAwareInterface;
+use Sylius\Component\Channel\Model\ChannelsAwareInterface;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+abstract class SlugUpdateHandler implements SlugUpdateHandlerInterface
+{
+    /** @var RedirectFactoryInterface */
+    private $redirectFactory;
```

Comment:
I would make all these protected, to allow better overriding.

Reply:
If I make all these private it opens up for BC breaks in the future. I love private :D But I still see your point. Wdyt?

Reply:
I often found myself doing a huge C/C from some vendor things just to override one small behaviour, and I think it's a very terrible DX.
Indeed, it opens up some BC breaks, but that's our role to think in long term and to be able to create method names that won't change every month.
I also understand your PoV, but still think it would be better to have them protected.

Reply:
I have made them protected now :) What I usually do in the case you describe is I open a PR with the change. Here's an example: https://github.com/Sylius/Sylius/pull/10756